client/components/Editor/utils/branches.js

```diff
@@ -59,8 +60,16 @@ export const mergeBranch = (sourceFirebaseRef, destinationFirebaseRef) => {
 				.child('merges')
 				.child(nextMergeKey)
 				.set(Object.values(changesSnapshotVal));
-			return Promise.all([setLastMergeKey, appendMerge]).then(() => {
-				return { mergeKey: nextMergeKey };
-			});
+
+			await Promise.all([setLastMergeKey, appendMerge]);
+
+			const { doc } = await getFirebaseDoc(
+				destinationFirebaseRef,
+				prosemirrorSchema,
+				nextMergeKey,
+			);
+			await storeCheckpoint(destinationFirebaseRef, doc, nextMergeKey);
```

Comment:
I might be missing a step in the logic, but won't building this checkpoint still fail and hit OOM/hanging issues for pubs that have 30k+ steps (just now at Merge creation rather than pub load)? That's still an improvement, as it doesn't let you publish something only to have it inaccessible. 

We could `getFirebaseDoc` based on the sourceFirebaseRef which would grab any relevant incremental checkpoint to handle the large pub case. Apologies if this is already doing that and I missed it.

Reply:
🤦 you're right, I meant to pass in `sourceFirebaseRef` here.